api/tests/unit/integrations/lead_tracking/hubspot/test_unit_hubspot_lead_tracking.py

```diff
@@ -1,469 +1,348 @@
-                "createdate": "2024-02-26T20:02:50.069Z",
-                "email": user.email,
-                "firstname": user.first_name,
-                "hs_all_contact_vids": "1000551",
-                "hs_email_domain": "example.com",
-                "hs_is_contact": "true",
-                "hs_is_unworked": "true",
-                "hs_marketable_status": user.marketing_consent_given,
-                "hs_object_id": "1000551",
-                "hs_object_source": "INTEGRATION",
-                "hs_object_source_id": "2902325",
-                "hs_object_source_label": "INTEGRATION",
-                "hs_pipeline": "contacts-lifecycle-pipeline",
-                "lastmodifieddate": "2024-02-26T20:02:50.069Z",
-                "lastname": user.last_name,
-            },
-            "properties_with_history": None,
-            "updated_at": datetime.datetime(2024, 2, 26, 20, 2, 50, 69000),
-        },
+        return_value={"id": HUBSPOT_USER_ID},
     )
-
-    assert getattr(organisation, "hubspot_organisation", None) is None
     # When
-    user.add_organisation(organisation, role=OrganisationRole.ADMIN)
-
+    tracker = HubspotLeadTracker()
+    hubspot_id = tracker.create_user_hubspot_contact(user)
     # Then
-    assert HubspotLead.objects.filter(hubspot_id=hubspot_lead_id).exists() is True
-    organisation.refresh_from_db()
-    assert organisation.hubspot_organisation.hubspot_id == future_hubspot_id
+    assert HubspotLead.objects.filter(hubspot_id=hubspot_id).exists() is True
+    assert hubspot_id == HUBSPOT_USER_ID
 
-    mock_create_company.assert_called_once_with(
-        name=organisation.name,
-        domain=domain,
-        active_subscription="free",
-        organisation_id=organisation.id,
+
+def test_create_organisation_lead_skips_all_tracking_when_lead_exists(
+    db: None,
+    organisation: Organisation,
+    mock_client_existing_contact: MagicMock,
+    mocker: MockerFixture,
+) -> None:
+    user = FFAdminUser.objects.create(
+        email="existing.lead@example.com",
+        first_name="Jane",
+        last_name="Doe",
+        marketing_consent_given=True,
+    )
+    HubspotLead.objects.create(user=user, hubspot_id=HUBSPOT_USER_ID)
+    HubspotOrganisation.objects.create(
+        organisation=organisation,
+        hubspot_id=HUBSPOT_COMPANY_ID,
     )
-    mock_create_contact.assert_called_once_with(user, future_hubspot_id)
-    mock_get_contact.assert_called_once_with(user)
+    tracker = HubspotLeadTracker()
+    tracker.create_organisation_lead(user=user, organisation=organisation)
 
+    mock_client_existing_contact.get_contact.assert_not_called()
+    mock_client_existing_contact.create_lead_form.assert_not_called()
+    mock_client_existing_contact.create_company.assert_not_called()
 
-def test_hubspot_with_filtered_email_domain_contact_and_new_organisation(
+
+def test_hubspot_find_company_by_domain_and_creates_organisation_lead_and_associates_existing_contact_from_hook(
```

Comment:
hmm... this test name is quite wordy and not really very descriptive - can we improve it at all? (perhaps some of the naming suggestions above factor in here too)